python/src/spark_rapids_ml/core.py

```diff
@@ -176,13 +176,13 @@ def _get_unwrapped_vec_cols(feature_col: Column) -> List[Column]:
     # is the array of active element values.
     # For dense vector, `type` field is 1, `size` and `indices` fields are None,
     # `values` field is the array of the vector element values.
+
+    feature_type = FloatType() if float32_input is True else DoubleType()
     return [
         features_unwrapped_vec_col.type.alias(alias.featureVectorType),
         features_unwrapped_vec_col.size.alias(alias.featureVectorSize),
         features_unwrapped_vec_col.indices.alias(alias.featureVectorIndices),
-        # Note: the value field is double array type, cast it to float32 array type
-        # for speedup following repartitioning.
-        features_unwrapped_vec_col.values.cast(ArrayType(FloatType())).alias(
+        features_unwrapped_vec_col.values.cast(ArrayType(feature_type)).alias(
```

Comment:
If float32_input = true can we avoid the redundant cast to double?

Reply:
Good catch!